### PR TITLE
Fix 2114: Sphinx copy-button plugin

### LINKED ISSUE #2114: Improve doc examples rendering
Currently we are rendering our examples in the docs like this https://pytorch.org/ignite/generated/ignite.engine.engine.Engine.html#ignite.engine.engine.Engine which is not very neat.
What we want to do is to find a better way to make it more good looking, and update our docs examples

image

- Replace "Examples" with some sphinx tag or something else, for example, like scikit-image does:

image

### PR DESCRIPTION
**Pull request by toxa23**
Fixes #2114

Description:
part 2 of 7: Added Sphinx plugin "copy-button" that allows copy code examples to clipboard

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Review by vfdev-5 — APPROVED** [review]
LGTM! Thanks @toxa23 

@ydcjeff could you please review this PR as well. Thanks

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 sphinx==3.4.3
 -e git+https://github.com/pytorch/pytorch_sphinx_theme.git#egg=pytorch_sphinx_theme
 sphinxcontrib-katex
+sphinx-copybutton==0.4.0
```
Comment (ydcjeff) [inline docs/requirements.txt:4]: Since this is a small extension, do we need to pin the version number?

File: docs/source/conf.py [hunk 2]
```diff
@@ -123,7 +124,7 @@
 html_static_path = ["_static", "_templates/_static"]
 
 html_context = {
-    "css_files": [
+    "extra_css_files": [
```
Comment (ydcjeff) [inline docs/source/conf.py:127]: Why is this change needed?

Comment (toxa23) [reply docs/source/conf.py:127]: Because `css_files` is a builtin context variable used by `app.add_css_file` in Sphinx' plugins setup. The copy button provides a css file. If we still want to overwrite `css_files` then we have to duplicate the copy button styles in `ignite_theme.css`.

File: docs/requirements.txt (hunk 1 above)
Comment (toxa23) [reply docs/requirements.txt:4]: At least is is guaranteed to work with pinned Sphinx version.

**Review by ydcjeff — APPROVED** [review]
LGTM

File: docs/requirements.txt (hunk 1 above)
Comment (oke-aditya) [reply docs/requirements.txt:4]: In my opinion you can set it to >=0.9.0, pytorch sphinx theme works with it fine. (Tested it for torchvision)

Comment (vfdev-5) [reply docs/requirements.txt:4]: @oke-aditya well, copy-button latest is 0.4.0. >=0.9.0 is gallery you are talking about : https://github.com/pytorch/vision/blob/b7c235892332c49a86c4be1b00fb6a672a3f9930/docs/requirements.txt#L4

Comment (oke-aditya) [reply docs/requirements.txt:4]: My bad. Sorry for that
